### ISSUE TITLE
Handle shaky network connectivity more resiliently

### ISSUE THREAD

**Issue by probonopd** [magne4000/quassel-webserver#257]
My reason for using this software is to be able to use IRC in situation with intermittent network connectivity, i.e., mobile, trains, etc.

When there is shaky connectivity, the web interface shows a red bar at the top (which is good). However, it asks me to reconnect (instead of retrying like.e.g, Gmail does) (which is slightly cumbersome), and then asks me to log in again (which is a real hassle), and shows a loading progress bar covering the entire screen (which makes it impossible to even look at the previous content).

I would love to see this be optimized more for semi-online/offline use cases, since I could imagine that an IRC web client is used by more people "on the road" than just me.

**Comment by probonopd** [comment]
Possibly [Service Workers](https://developers.google.com/web/fundamentals/primers/service-workers/) could be used to facilitate this.

**Comment by magne4000** [comment]
That's a really complicated problem here, that couldn't simply be solved, because it's closely tied to how current Quassel protocol works, which isn't really mobile/web friendly.
When connection is lost for any reason, we are forced to start the connection process again from the beginning.

The current reload + login part is cumbersome I agree, and this could be improved. I could just try to reconnect indefinitely without the need to refresh the page.

**Comment by probonopd** [comment]
> I could just try to reconnect indefinitely without the need to refresh the page.

Yes, I think this would already be a big improvement, together with a local caching of the already-downloaded queues in the browser.
